R/run-model.R

```diff
@@ -317,8 +317,23 @@ naomi_prepare_data <- function(data, options) {
   calendar_quarter_t1 <- options$calendar_quarter_t1
   calendar_quarter_t2 <- options$calendar_quarter_t2
   calendar_quarter_t3 <- options$calendar_quarter_t3
-  calendar_quarter_t4 <- options$calendar_quarter_t4
-  calendar_quarter_t5 <- options$calendar_quarter_t5
+
+  if(is.null(options$calendar_quarter_t4)) {
+    # If T4 is not specified, set T4 to 24-months after T3
+    t4 <- calendar_quarter_to_quarter_id(calendar_quarter_t3) + 6
+    calendar_quarter_t4 <- quarter_id_to_calendar_quarter(t4)
+  } else{
+    calendar_quarter_t4 <- options$calendar_quarter_t4
+  }
+
+  if(is.null(options$calendar_quarter_t5)) {
+    # If T5 is not specified, set T4 to 36-months after T3
+    t5 <- calendar_quarter_to_quarter_id(calendar_quarter_t3) + 9
```

Comment:
Thanks—in a pathological situation, this could create an issue if T4 was set but T5 was not and T4 was more than 9 months after T3.

That _shouldn't_ ever happen.  But maybe just for completeness set this 3 months after T4?

(This would still leave open a potential issue in a situation where T5 was set to a time less than 6 months after T3, but T4 was not set. But maybe just leave that as an error.)

Reply:
@jeffeaton That makes sense - have set T5 relative to T4.